### ISSUE TITLE
npm start - from master zip - fails to compile

### ISSUE THREAD

**Issue by goldel777** [ckeditor/ckeditor5-angular#207]
**npm start**

> @ckeditor/ckeditor5-angular@1.2.3 start C:\Users\goldlo5\git\ckeditor5-angular-master
> ng serve --open

    WARNING: Zone.js does not support native async/await in ES2017.
    These blocks are not intercepted by zone.js and will not triggering change detection.
    See: https://github.com/angular/zone.js/pull/1140 for more information.
10% building 3/3 modules 0 activei ｢wds｣: Project is running at http://localhost:4200/webpack-dev-server/
i ｢wds｣: webpack output is served from /
i ｢wds｣: 404s will fallback to //index.html

chunk {main} main.js, main.js.map (main) 2.02 kB [initial] [rendered]
chunk {polyfills} polyfills.js, polyfills.js.map (polyfills) 154 kB [initial] [rendered]
chunk {runtime} runtime.js, runtime.js.map (runtime) 6.15 kB [entry] [rendered]
chunk {styles} styles.js, styles.js.map (styles) 9.72 kB [initial] [rendered]
chunk {vendor} vendor.js, vendor.js.map (vendor) 340 kB [initial] [rendered]
Date: 2020-06-02T13:46:15.897Z - Hash: 5833d29d3e0c048bbb7c - Time: 20714ms

ERROR in node_modules/zone.js/dist/zone.js.d.ts:600:21 - error TS2503: Cannot find namespace 'NodeJS'.

600 declare var global: NodeJS.Global;
                        ~~~~~~
src/app/context-demo/context-demo.ts:3:27 - error TS2307: Cannot find module '../../../ckeditor/build/cksource'.

3 import * as CKSource from '../../../ckeditor/build/cksource';
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/app/demo-form/demo-form.component.ts:8:27 - error TS2307: Cannot find module '../../../ckeditor/build/cksource'.

8 import * as CKSource from '../../../ckeditor/build/cksource';
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/app/demo-reactive-form/demo-reactive-form.component.ts:7:27 - error TS2307: Cannot find module '../../../ckeditor/build/cksource'.

7 import * as CKSource from '../../../ckeditor/build/cksource';
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/app/simple-usage/simple-usage.component.ts:2:27 - error TS2307: Cannot find module '../../../ckeditor/build/cksource'.

2 import * as CKSource from '../../../ckeditor/build/cksource';
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/app/watchdog-demo/watchdog-demo.ts:3:27 - error TS2307: Cannot find module '../../../ckeditor/build/cksource'.

3 import * as CKSource from '../../../ckeditor/build/cksource';
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **
i ｢wdm**｣: Failed to compile.**

**Comment by goldel777** [comment]
It looks like for the for issue below:

ERROR in node_modules/zone.js/dist/zone.js.d.ts:600:21 - error TS2503: Cannot find namespace 'NodeJS'.


The solution is to add the following to tsconfig.app.json


{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "./out-tsc/app",
    **"types": ["node"]**
  },
  "include": [
    "src/**/*.ts"
  ],
  "exclude": [
    "src/test.ts",
    "src/**/*.spec.ts"
  ]
}

**Comment by ma2ciek** [comment]
Hi!

I guess that's a minor issue actually. The `Zone.js` package should work outside of the Node.js environment So IDK why it depends on typings for the server-side JS.

You missed this script: `npm run build-ckeditor` before building the editor.

**Comment by goldel777** [comment]
Thanks that took care of the issue with missing modules.   :)



When I updated the types: "nodes", I got the following.  Which I corrected by downgrading 

npm install @types/node@14.0.4 --saveDev

- see stacktrace url below

ERROR in node_modules/@types/node/ts3.5/globals.global.d.ts:1:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'global' must be of type 'Global', but here has type 'Global & typeof globalThis'.

1 declare var global: NodeJS.Global & typeof globalThis;
              ~~~~~~

  node_modules/zone.js/dist/zone.js.d.ts:600:13
    600 declare var global: NodeJS.Global;
                    ~~~~~~
    'global' was also declared here.
src/app/context-demo/context-demo.ts:3:27 - error TS2307: Cannot find module '../../../ckeditor/build/cksource'.

https://stackoverflow.com/questions/62054471/unable-to-build-since-upgrading-from-angular-9-17-to-9-19
